sao/mappings/mapping.py

```diff
@@ -0,0 +1,146 @@
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+class Mapping(ABC):
+    '''
+    Approximation is a function mapping f: R^n -> R^n
+    '''
+
+    @property
+    def name(self):
+        return self.__class__.name
+
+    def __init__(self, mapping=None):
+        self.map = mapping
+        if self.map is None:
+            self.map = Empty()
+
+    def update(self, x, f, df, ddf=None):
+        """
+        This method updates the approximation instance.
+
+        :param x: Current design
+        :param f: A vector of size [m+1] that holds the response values at the current design -x-
+        :param df: A matrix of size [m+1, n] that holds the sensitivity values at the current design -x-
+        :param ddf: Optionally get the 2nd-order sensitivity array
+        :return: self: For method cascading
+        """
+        self.map.update(x, f, df, ddf)
+        self._update(x, f, df, ddf)
+
+    def clip(self, x):
+        return self._clip(self.map.clip(x))
+
+    def g(self, x):
+        '''Chain rule'''
+        return self._g(self.map.g(x))
+
+    def dg(self, x):
+        '''Chain rule first derivative'''
+        return self._dg(self.map.g(x)) * self.map.dg(x)
+
+    def ddg(self, x):
+        '''Chain rule second derivative'''
+        return self._ddg(x) * self.map.dg(x) ** 2 + self._dg(x) * self.map.ddg(x)
+
+    def _update(self, x, f, df, ddf=None):
+        pass
+
+    def _clip(self, x):
+        return x
+
+    def _g(self, x):
+        return x
+
+    def _dg(self, x):
+        return np.ones_like(x)
+
+    def _ddg(self, x):
+        return np.zeros_like(x)
+
+
+class Empty(Mapping):
+
+    def __init__(self):
+        pass
+
+    def update(self, x, f, df, ddf=None):
+        pass
+
+    def clip(self, x):
+        return x
+
+    def g(self, x):
+        return x
+
+    def dg(self, x):
+        return np.ones_like(x)
+
+    def ddg(self, x):
+        return np.zeros_like(x)
```

Comment:
```suggestion
class Empty(Mapping):
    ...
```

It seems behaviour of `Empty()` is the same as the base `Mapping`